pkg/component/component.go

```diff
@@ -23,9 +23,15 @@ import (
 // GenerateMonitoringCfgFn is a function that can inject information into the model generation process.
 type GenerateMonitoringCfgFn func(map[string]interface{}, []Component, map[string]string) (map[string]interface{}, error)
 
+type HeadersProvider interface {
+	Headers() map[string]string
+}
```

Comment:
I wonder if this interface is really needed, a `map[string]string` is pretty standard for headers. Anyway, it isn't a blocker

Reply:
i did this so i can pass agentinfo object but test with custom thingy. 
i dont want to pass headers as they are, but rather something which provides me up to date flags.